File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Nodejs/utils/RuntimeLogger.js
const os = require('os');
const process = require('process');

class RuntimeLogger {
    static notLoggedYet = true;

    static getRuntimeInfo() {
        try {
            return `JavaScript Managed Runtime Info:\n` +
                `Node.js Version: ${process.version}\n` +
                `OS Version: ${os.type()} ${os.release()}\n` +
                `Process Architecture: ${os.arch()}\n` +
                `Current Directory: ${process.cwd()}\n`;
        } catch (e) {
            return "JavaScript Managed Runtime Info: Error while fetching runtime info";
        }
    }

    static printRuntimeInfo() {
        if (this.notLoggedYet) {
            console.log(this.getRuntimeInfo());
            this.notLoggedYet = false;
        }
    }
}

module.exports = RuntimeLogger;

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Python/javonet/utils/RuntimeLogger.py
import platform
import os
import sys


class RuntimeLogger:

    not_logged_yet = True
    @staticmethod
    def get_runtime_info():
        try:
            return (
                "Python Managed Runtime Info:\n"
                f"Python Version: {platform.python_version()}\n"
                f"Python executable path: {sys.executable}\n"
                f"Python Path: {sys.path}\n"
                f"Python Implementation: {platform.python_implementation()}\n"
                f"OS Version: {platform.system()} {platform.version()}\n"
                f"Process Architecture: {platform.machine()}\n"
                f"Current Directory: {os.getcwd()}\n"
            )
        except Exception as e:
            return "Python Managed Runtime Info: Error while fetching runtime info"

    @staticmethod
    def print_runtime_info():
        if RuntimeLogger.not_logged_yet:
            print(RuntimeLogger.get_runtime_info())
            RuntimeLogger.not_logged_yet = False
    
